app/src/main/java/com/dmonzonis/nookpendium/CollectionActivity.kt

```diff
@@ -30,5 +34,29 @@ class CollectionActivity : AppCompatActivity() {
         recyclerView.setHasFixedSize(true)
         recyclerView.layoutManager = viewManager
         recyclerView.adapter = viewAdapter
+
+        tabLayout.addOnTabSelectedListener(object: TabLayout.OnTabSelectedListener {
+            override fun onTabReselected(p0: TabLayout.Tab?) {
+            }
+
+            override fun onTabUnselected(p0: TabLayout.Tab?) {
+            }
+
+            override fun onTabSelected(tab: TabLayout.Tab?) {
+                val tabName = tab!!.text.toString()
+                val filename : String
+                if (tabName == getString(R.string.fish)) {
+                    filename = "fish.xml"
```

Comment:
There are also string resources for the file names (`R.string.fish_file` and so on).
You could also use Kotlin's pattern matching to make this look a bit better:

```Kotlin
val filename = when(tabName) {
    getString(R.string.fish) -> getString(R.string.fish_file)
    getString(R.string.insects) -> getString(R.string.insects_file)
    else -> getString(R.string.underwater_file)
}
```

Although I think it may be more elegant use some identifier for the tab or an enum and make that match against the filename, instead of using the tab's text directly.